Composer/packages/tools/language-servers/language-generation/src/LGServer.ts

```
@@ -0,0 +1,355 @@
+      this.cleanPendingValidation(event.document);
+      this.cleanDiagnostics(event.document);
+    });
+
+    this.connection.onInitialize(params => {
+      if (params.rootPath) {
+        this.workspaceRoot = URI.file(params.rootPath);
+      } else if (params.rootUri) {
+        this.workspaceRoot = URI.parse(params.rootUri);
+      }
+      this.connection.console.log('The server is initialized.');
+      return {
+        capabilities: {
+          textDocumentSync: this.documents.syncKind,
+          codeActionProvider: false,
+          completionProvider: {
+            resolveProvider: true,
+          },
+          hoverProvider: true,
+          foldingRangeProvider: false,
+        },
+      };
+    });
+    this.connection.onCompletion(params => this.completion(params));
+    this.connection.onHover(params => this.hover(params));
+
+    this.connection.onRequest((method, params) => {
+      if (InitializeDocumentsMethodName === method) {
+        const { uri, content, template } = params;
+        this.LGDocuments.push({ uri, content, template });
+        // run diagnostic
+        const textDocument = this.documents.get(uri);
+        if (textDocument) {
+          setTimeout(() => {
+            this.validate(textDocument);
+          }, 500);
+        }
+      }
+    });
+  }
+
+  start() {
+    this.connection.listen();
+  }
+
+  protected getLGDocument(document: TextDocument): LGDocument | undefined {
+    const LGDocument = this.LGDocuments.find(item => item.uri === document.uri);
+    return LGDocument;
+  }
+  protected getLGDocumentContent(document: TextDocument): string {
+    const LGDocument = this.LGDocuments.find(item => item.uri === document.uri);
+    let text = document.getText();
+    if (LGDocument) {
+      // concat new content
+      const { content, template } = LGDocument;
+      try {
+        text = updateTemplateInContent(content, template);
+        // eslint-disable-next-line no-empty
+      } catch (error) {}
```

Be careful when silencing errors, could easily lead to data corruption and loss of user data.